services/imageproxy/local.go

```diff
@@ -78,6 +81,19 @@ func (backend *LocalBackend) GetImage(w http.ResponseWriter, r *http.Request, im
 		return
 	}
 
+	u, err := url.Parse(imageURL)
+	if err != nil {
+		mlog.Error("Failed to parse URL for proxied image", mlog.String("url", imageURL), mlog.Err(err))
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte{})
+		return
+	}
+
+	extension := strings.ToLower(filepath.Ext(u.Path))
+	if mime.TypeByExtension(extension) == "image/svg+xml" {
```

Comment:
The proxy backend primarily uses the `Content-Type` response header for MIME type comparisons, so there's a potential mismatch here: a malicious or otherwise misconfigured image server can serve `/image.jpg` with `Content-Type: image/svg+xml`.

A few alternative ideas on how to fix it:

 - Use the `GetImageDirect` implementation instead of `GetImage` -- likely comes with a performance hit since responses would then be buffered
 - Wrap the `http.ResponseWriter` in a custom implementation of the interface that allows us to conditionally set a `Content-Disposition` at the same time as the `Content-Type` value is being copied over
 - Fork the proxy backend and add the `Content-Disposition` header there
 - Somehow enforce the content type detected here and overwrite the one detected by the backend

I'm 0/5 on how feasible any of these are, though. If setting the `Content-Disposition` header conditionally turns out too complicated, we might want to set it on all responses instead, but that probably requires explicit PM approval.

Reply:
I have been thinking about 2. But the problem seems to be that the ResponseWriter wrapper only allows us to hook into WriteHeader and Write calls. I'll try and see if I can get the value of the Header within these calls, and if so, modify it.

I think 1 would not be very ideal and I would consider it after all the other alternatives.

Reply:
Nvm .. found it. PTAL.